nixos/modules/system/activation/switch-to-configuration.pl

```diff
@@ -18,6 +18,12 @@
 my $restartListFile = "/run/systemd/restart-list";
 my $reloadListFile = "/run/systemd/reload-list";
 
+# Parse restart/reload requests by the activation script
+my $restartByActivationFile = "/run/systemd/activation-restart-list";
+my $reloadByActivationFile = "/run/systemd/activation-reload-list";
+my $dryRestartByActivationFile = "/run/systemd/dry-activation-restart-list";
+my $dryReloadByActivationFile = "/run/systemd/dry-activation-reload-list";
```

Comment:
How about not fiddling with systemd's internal runtime directory?

```suggestion
my $restartByActivationFile = "/run/activation-restart-list";
my $reloadByActivationFile = "/run/activation-reload-list";
my $dryRestartByActivationFile = "/run/dry-activation-restart-list";
my $dryReloadByActivationFile = "/run/dry-activation-reload-list";
```

This might also work:

```suggestion
my $restartByActivationFile = "/run/nixos/activation-restart-list";
my $reloadByActivationFile = "/run/nixos/activation-reload-list";
my $dryRestartByActivationFile = "/run/nixos/dry-activation-restart-list";
my $dryReloadByActivationFile = "/run/nixos/dry-activation-reload-list";
```

Reply:
We doing this right now with the old logic, but it is not very clean.

Reply:
I changed both the new and the old logic to /run/nixos